### PR TITLE
Fix doc for initializer

### PR DESCRIPTION
**Pull request by bddppq**
https://github.com/onnx/onnx/issues/511

### CODE REVIEW

File: onnx/onnx-ml.proto [hunk 1]
```diff
@@ -236,13 +236,10 @@ message GraphProto {
   // The name of the graph.
   optional string name = 2;   // namespace Graph
 
-  // A list of named tensor values (constants), used to specify default
+  // A list of named tensor values (constants), used to specify constant
```
Comment (houseroad) [inline onnx/onnx-ml.proto]: delete the first `constants` in the parenthesis?

File: onnx/onnx-ml.proto [hunk 2]
```diff
@@ -236,13 +236,10 @@ message GraphProto {
...
   // values for some of the inputs of the graph.
   // Each TensorProto entry must have a distinct name (within the list) that
   // also appears in the input list.
-  // In an evaluation, the default value specified here is used if and only if
```
Comment (houseroad) [inline onnx/onnx-ml.proto]: I think this paragraph says that even the graph contains some values in the initializer, we can still ignore it by passing the value in the input list.

Comment (bddppq) [reply onnx/onnx-ml.proto]: @houseroad Yeah that was in the old days we were trying to provide the convenience to users to override the constant values in the `run_model`, `run_node` python interface, but it doesn't work anymore as we have more and more stuffs relying on the static graph analysis.

File: onnx/onnx-ml.proto (hunk 1 above)
Comment (houseroad) [inline onnx/onnx-ml.proto]: Also shall we explicitly say these are weights of the models. :-)

**Comment by linkerzhang** [comment]
Should we allow "GraphProto.input includes both weights and graph input," please? I feel graphproto.input should only contain graph input. make sense please?

**Comment by bddppq** [comment]
@linkerzhang the purpose of putting initializer into graph.input is if (someday) onnx supports training,  then existing models can simply drop the initializers and its graph is still a valid graph. TBH I don't feel there are too much differences between including or not including initilaizers' ValueInfoProto in graph.input :-) But since we have started this way, it's probably not worth the effort of introducing a breaking change?

**Comment by gramalingam** [comment]
@bddppq : but the change in the documentation seems to have dropped the lines that say that the caller-specified value overrides the weight specified in the model ... so, this is a bit confusing. If we want to retain the original behavior, it would be nice to have a line in the documentation indicating this.

**Pull request merged**